tests/behat/validate_rating.feature

```diff
@@ -0,0 +1,95 @@
+@mod @mod_ratingallocate
+Feature: When a student attempts to rate choices it should be validated prior to changing.
+
+  Background:
+    Given the following "courses" exist:
+      | fullname | shortname | category | groupmode |
+      | Course 1 | C1        | 0        | 1         |
+    And the following "users" exist:
+      | username | firstname | lastname | email |
+      | teacher1 | Teacher | 1 | teacher1@example.com |
+      | student1 | Student | 1 | student1@example.com |
+    And the following "course enrolments" exist:
+      | user | course | role |
+      | teacher1 | C1 | editingteacher |
+      | student1 | C1 | student |
+    And I log in as "teacher1"
+    And I am on "Course 1" course homepage with editing mode on
+    And I add a "Fair Allocation" to section "1" and I fill the form with:
+      | id_name | Validated Rating |
+      | strategy    | strategy_points |
+      | accesstimestart[day] | ##2 days ago##j## |
+      | accesstimestart[month] | ##2 days ago##n## |
+      | accesstimestart[year] | ##2 days ago##Y## |
+      | strategyopt[strategy_points][maxzero] | 2 |
+    And I follow "Validated Rating"
+    And I press "Edit Choices"
+    And I add a new choice with the values:
+      | title       | My first choice |
+      | explanation | Test 1          |
+      | maxsize     |	2	    	  |
+    And I add a new choice with the values:
+      | title       | My second choice |
+      | explanation | Test 2           |
+      | maxsize     |	2	    	   |
+    And I add a new choice with the values:
+      | title       | My third choice |
+      | explanation | Test 3  		  |
+      | maxsize     |	2	    	  |
+    And I add a new choice with the values:
+      | title       | My fourth choice |
+      | explanation | Test 4  		  |
+      | maxsize     |	2	    	  |
+    And I log out
+
+  Scenario: The user cannot enter values less than 0.
+    When I log in as "student1"
+    And I am on "Course 1" course homepage
+    And I follow "Validated Rating"
+    And I press "Edit Rating"
+    And I rate choices with the following points:
+      | My first choice | -1 |
+      | My second choice | 1 |
+      | My third choice | 1 |
+      | My fourth choice | 99 |
+    And I press "Save changes"
+    Then I should see "The points that you assign to a choice must be between 0 and 100."
+
+  Scenario: The user cannot enter values higher than the (default) maximum.
+    When I log in as "student1"
+    And I am on "Course 1" course homepage
+    And I follow "Validated Rating"
+    And I press "Edit Rating"
+    And I rate choices with the following points:
+      | My first choice | -1 |
```

Comment:
You might want to change -1 to 1 here.

Reply:
No, because then I get a different validation message that tells me that points must sum up to 100.

Reply:
Removed as per our oral discussion: In the end, this test is rather useless as it will trigger the expected message already because of the negative rating. And that has already been checked in a different test.